src/controllers/url-controller.ts

```diff
@@ -0,0 +1,152 @@
+import { Request, Response } from 'express';
+import prisma from '../libs/prisma-client';
+
+import { convertToHyphenated, generateShortUrl } from '../utils/short-url-generator';
+import logger from '../logs/logger';
+
+const successMsg = 'Url successfully fetched';
+const notFoundMsg = 'No Url found!';
+
+//  CREATE A SHORTEN URL FROM A LONG URL
+export const createUrl = async (req: Request, res: Response) => {
+  const { url, customName } = req.body;
+
+  if (!url) return res.status(400).send({ message: 'Url is required!' });
+  try {
+    const existingUrl = await prisma.url.findFirst({
+      where: {
+        originalUrl: url,
+      },
+    });
+    if (existingUrl) {
+      return res.status(400).send({ message: 'Url already exists!' });
+    }
+
+    const urlString = generateShortUrl();
+
+    const custom = customName && convertToHyphenated(customName);
+
+    const originalUrl = url;
+    const shortUrl = custom ? `https://shortit/${custom}` : `https://shortit/${urlString}`;
+    const customNameUrl = custom ? `${custom}` : '';
+
+    const shortenUrl = await prisma.url.create({
+      data: {
+        originalUrl,
+        shortUrl,
+        customName: customNameUrl,
+      },
+    });
+    return res.status(201).json({ message: successMsg, shortenUrl });
+  } catch (error) {
+    logger.info(error);
+    return res.status(400).send({ message: 'Error adding url' });
```

Comment:
Change this in all the handlers

```suggestion
    logger.error(error);
    return res.status(500).send({ message: 'Error adding url' });
```